ksqldb-engine/src/main/java/io/confluent/ksql/serde/InternalFormats.java

```diff
@@ -45,22 +46,23 @@ private InternalFormats() {
    * tracks the removal.
    *
    * @param keyFormat key format.
-   * @param valueFormat value format.
+   * @param valueFormatInfo value format info.
    * @return Formats instance.
    * @see <a href=https://github.com/confluentinc/ksql/issues/6296>Issue 6296</a>
    * @see SerdeFeaturesFactory#buildInternal
    */
-  public static Formats of(final KeyFormat keyFormat, final FormatInfo valueFormat) {
+  public static Formats of(final KeyFormat keyFormat, final FormatInfo valueFormatInfo) {
     // Do not use NONE format for internal topics:
     if (keyFormat.getFormatInfo().getFormat().equals(NoneFormat.NAME)) {
       throw new IllegalArgumentException(NoneFormat.NAME + " can not be used for internal topics");
     }
 
     return Formats.of(
-        keyFormat.getFormatInfo(),
-        valueFormat,
+        keyFormat.getFormatInfo().copyWithoutProperty(ConnectProperties.SCHEMA_ID),
```

Comment:
It'd be good to add a quick comment here to explain why SCHEMA_ID is being removed. Something like "Internal formats should not use user-specified schema IDs" (or whatever you think it clearest).